compiler/passes/scopeResolve.cpp

```diff
@@ -250,12 +250,33 @@ static void markGenerics() {
     } while (changed);
 }
 
+static void checkClass(AggregateType* ct) {
+  if (isClass(ct) && ct->symbol->hasFlag(FLAG_EXTERN)) {
+    USR_FATAL_CONT(ct, "Extern classes are not supported.");
+  }
+  // Warn for superclass should be marked generic
+  // Error for a concrete superclass that is marked generic
+  if (isClass(ct) && ct->dispatchParents.n == 1) {
+    if (AggregateType* parent = ct->dispatchParents.v[0]) {
+      if (isClass(parent)) {
+        if (!ct->symbol->hasFlag(FLAG_SUPERCLASS_MARKED_GENERIC) &&
+            parent->isGeneric() && !parent->isGenericWithDefaults()) {
+          USR_WARN(ct->symbol, "inherits expr should include (?) for a generic superclass");
+        }
+        if (ct->symbol->hasFlag(FLAG_SUPERCLASS_MARKED_GENERIC) &&
+            !parent->isGeneric()) {
+          USR_FATAL(ct->symbol, "inherits expr should not include (?) for a concrete class");
```

Comment:
Following Brad's suggestion, switch these to talking about the "parent class", ex.

`warning: missing '(?)' after a generic parent class`
`error: '(?)' after a concrete parent class`